### DISCUSSION TITLE
Cannot make true-myth work with jest

### DISCUSSION THREAD

**Discussion by gegenschall** [true-myth/true-myth#449] [General]
I'm trying to use true-myth in my project but cannot make it work with jest. The error I cannot work around is:

```
Details:

    /path/to/project/node_modules/true-myth/dist/result.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){import Unit from './unit.js';
                                                                                      ^^^^^^

    SyntaxError: Cannot use import statement outside a module

      2 | import { InjectRepository } from '@mikro-orm/nestjs';
      3 | import { Injectable } from '@nestjs/common';
    > 4 | import { Result, err, ok } from 'true-myth/result';
        | ^
```

I tried a couple of things:
- Enable ESM and use true-myth 6, however that's mostly impossible to use in my project so sadly not an option.
- Fiddling around with tsconfig.json, trying `allowJs` and similar config options.
- I'm using ts-jest but did install jest-babel, added `transform` and `transformIgnorePatterns`.

I have a monorepo with yarn 3 set up and my directory structure looks like this:
```
|-- services/
           |-- serviceA
                     |-- src/
                     |-- jest.config.js
                     |-- tsconfig.json
                     |-- ...
           |-- serviceB
|-- jest.config.js
|-- tsconfig.json
```

Each `jest.config.js` in subdirectories is mostly just this:
```
/* eslint-disable @typescript-eslint/no-var-requires */
const rootConfig = require('../../jest.config');

module.exports = {
  rootDir: 'src',
  ...rootConfig,
};
```

Is there any way to make true-myth work with jest?

**Comment by chriskrycho** [comment]
Clarification: Are you working with v5, and finding *that* doesn’t work?

**Reply by gegenschall** [reply]
Yes, that's correct. Using `5.4.0`

**Comment by gegenschall** [comment]
I did some further digging: true-myth 5.x makes use of `import` statements in its compiled .js files. Thus, to use it I think I have two options:

1. use ESM in my app which isn't an option for different reasons, or
2. use dynamic `import()` statements, which isn't an option either because these have to be `await`-ed and that can't be done top-level without using ESM support in node. If I'm not able to import top-level in a `.ts` file I can't really annotate types in functions in that file.

Apologies if I'm looking at this completely wrong but I'm a little confused on how to use it.

**Reply by chriskrycho** [reply]
Sorry for the very long delay; that’s very odd. Did you get this resolved or just move on? v5 definitely should have worked that way.

**Comment by kevinjhanna** [comment]
@chriskrycho same issue here. 
I'm running `v5.4.0`, and finding problems when running my tests with Jest:


When the import is:
```
import Result from 'true-myth/result';
```

Issue:
```
    .../node_modules/true-myth/dist/result.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){import Unit from './unit.js';
                                                                                      ^^^^^^
    SyntaxError: Cannot use import statement outside a module
```

When the import is:
```
import { Result } from 'true-myth';
```

Issue is:
```
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){export { Maybe } from './maybe.js';
                                                                                      ^^^^^^
    SyntaxError: Unexpected token 'export'
    ```

**Reply by chriskrycho** [reply]
Following up: does the recent v7 release solve this?
